### ISSUE TITLE
Add option to use shadow.test.node without process exit

### ISSUE THREAD

**Issue by jpmonettas** [thheller/shadow-cljs#1202]
I'm working in a test coverage tool for ClojureScript and would be great to be able to call the code in shadow.test.node without it exiting the process after finishing. 

There is a FIXME in the code already for it : https://github.com/thheller/shadow-cljs/blob/master/src/main/shadow/test/node.cljs#L9

**Comment by thheller** [comment]
I'd recommend just copying that namespace and making your own adjustments. Given that you'll likely need more modifications than just this do get the coverage stuff working, that seems like the easiest path. You can plug in your own namespace via the `:runner-ns` option in the build config.

**Comment by jpmonettas** [comment]
Yeah | have been doing that so far. Since it is working with just running the test without exiting the process and I saw the FIXME there I created the issue, but I'm ok also replicating the code since I don't think that is going to change much.

**Comment by thheller** [comment]
I'm fine with adding that option. I'm just unsure how things would work if there is suddenly no exit code anymore.

**Comment by jpmonettas** [comment]
If it can be optional and on by default, then what happens when someone disables it would be on their sides.
Maybe the process exit can also be done on a different place for improved DX? I find tricky that by loading a namespace suddenly your process exit after running the tests, which I found hard to figure out.

**Comment by thheller** [comment]
I guess thats true. I could change it so it only exists when actually running via the `main` fn? Then just loading the ns doesn't immediately change test behavior. Would that suffice or would you still need that option?

**Comment by jpmonettas** [comment]
Yeah, that would be great.

**Comment by jpmonettas** [comment]
Oh thanks for the detailed explanation. I'm currently copying all `shadow.test.node` and then using it as `:runner-ns` and it seams to be working fine.
 
What I need  for the test coverage  tool is pretty straight forward, which is to run something before and after tests. The before to initialize some tracing structures, and the after for generating the reports.

Just to be clear the coverage tool is already working with the browser and nodejs with the current version of shadow-cljs. I opened this issue because I saw the fixme on `shadow.test.node` and I thought that if there was a way of preventing that then I don't need to copy/maintain the entire namespace.
